src/create.ts

```diff
@@ -3,14 +3,27 @@
 import {
   createContext,
   useContext,
-  useEffect,
   useMemo,
   useReducer,
   useRef,
+  useLayoutEffect,
+  useEffect
 } from 'react';
 import {Action, Dispatch, Store} from 'redux';
 import shallowEqual from './shallowEqual';
 
+// React currently throws a warning when using useLayoutEffect on the server.
+// To get around it, we can conditionally useEffect on the server (no-op) and
+// useLayoutEffect in the browser. We need useLayoutEffect to ensure the store
+// subscription callback always has the selector from the latest render commit
+// available, otherwise a store update may happen between render and the effect,
```

Comment:
> a store update may happen between render and the effect, which may cause missed updates

Are you sure this is true? This is precisely why we check for updates after subscribing on line 129 below.

> otherwise a store update may occur before the subscription is created and an inconsistent state may be observed

Again, won't the updates check handle this?

Reply:
yea, useLayoutEffect not about subscriptions